server/content.js

```diff
@@ -2,14 +2,56 @@ const fse = require('fs-extra');
 const fs = require('fs');
 const path = require('path');
 
+const fileExistError = new Error('This slug name is already taken');
+
 const save = (contentType, contentName, data, cb) => {
-  console.log(data);
+  const parsedData = JSON.parse(String(data));
+  const filePath = path.resolve('_content', contentType, contentName + '.json');
+  const newFilePath = path.resolve('_content', contentType, parsedData.slug + '.json');
+
+  if (contentName === parsedData.slug) {
+    fse.outputFile(newFilePath, data, cb);
+  } else {
+    fs.stat(newFilePath, function(err, stat) {
+      if (err === null) {
+        throw fileExistError;
+      } else if (err.code === 'ENOENT') {
+        fs.rename(filePath, newFilePath, err => {
+          if (err) throw err;
+          fse.outputFile(newFilePath, data, cb);
+        });
+      } else {
+        console.error(err.code);
+        throw err;
```

Comment:
thrown errors should be handled on the API level as well to pass related HTTP error code back to browser. https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_errors

Reply:
So which one should we use here ?